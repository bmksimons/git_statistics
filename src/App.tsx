import React from "react";
import "./App.css";
import Commits from "./Commits";
import Issues from "./Issues";
import Header from "./Header";

function App() {
  return (
    <div className="App">
      <Header />
      <Issues />
      <Commits />
    </div>
  );
}

export default App;
