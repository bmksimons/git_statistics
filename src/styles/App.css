@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');

.App {
  text-align: center;
  background-color: #dbe2e5;
  display: flex;
  flex-direction: column;
  height: 100vh;
}

body {
  font-family: 'Montserrat', sans-serif;
}
