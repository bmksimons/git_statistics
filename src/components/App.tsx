import React from "react";
import "../styles/App.css";
import Header from "./Header";
import Issues from "./Issues";
import Commits from "./Commits";

function App() {
  return (
    <div className="App">
      <Header />
      <Issues />
      <Commits />
    </div>
  );
}

export default App;
