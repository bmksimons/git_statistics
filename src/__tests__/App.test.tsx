import { render, screen } from '@testing-library/react';
import App from '../components/App';

test("renders", () => {
  const { container } = render(<App/>);
  expect(container).toMatchSnapshot();
});
