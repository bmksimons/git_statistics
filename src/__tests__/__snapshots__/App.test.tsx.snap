// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <h1>
        GitLab Statistics
      </h1>
    </div>
    <div
      class="content-wrapper"
    >
      <div
        class="sidebar-wrapper"
      >
        <div
          class="sidebar"
        >
          <nav>
            <ul>
              <li>
                Commits
              </li>
              <li>
                Issues
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class="wrapper"
        >
          <h2
            class="component-header"
          >
            Issues
          </h2>
          <p
            class="component-description"
          >
            Overview over all issues from the project, with the ability to filter between showing all, opened or closed issues. An issue has an issuenumber, title and labels. If you click on an issue the description will show up in a pop-up.
          </p>
          <div
            class="selectFilter"
          >
            <select
              data-testid="selectFilterIssue"
              id="selectFilterIssue"
            >
              <option
                disabled=""
                hidden=""
                value=""
              >
                Show All
              </option>
              <option
                value="all"
              >
                Show all
              </option>
              <option
                value="opened"
              >
                Show open
              </option>
              <option
                value="closed"
              >
                Show closed
              </option>
            </select>
          </div>
          <div
            class="issueList"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
